contribs/src/main/java/com/netflix/conductor/contribs/queue/kafka/KafkaObservableQueue.java

```diff
@@ -0,0 +1,519 @@
+				logger.debug("Producer Record: key {}, value {}, partition {}, offset {}", record.key(), record.value(),
+						metadata.partition(), metadata.offset());
+			} catch (InterruptedException e) {
+				Thread.currentThread().interrupt();
+				logger.error("Publish message to kafka topic {} failed with an error: {}", queueName, e.getMessage(),
+						e);
+			} catch (ExecutionException e) {
+				logger.error("Publish message to kafka topic {} failed with an error: {}", queueName, e.getMessage(),
+						e);
+				throw new ApplicationException(ApplicationException.Code.INTERNAL_ERROR, "Failed to publish the event");
+			}
+		}
+		logger.info("Messages published to kafka topic {}. count {}", queueName, messages.size());
+
+	}
+
+
+	@Override
+	public void processFailures(List<Message> messages, EventProcessingFailures failures) {
+		processFailures(messages, failures.getTransientFailures());
+		processFailures(messages, failures.getFailures());
+	}
+
+	public void processFailures(List<Message> messages, List<EventExecution> failures) {
+		final String errorQueueName = this.queueName + "-errors";
+		failures.forEach(failure -> {
+			Optional<MessageEventFailure> messageEventFailure = getMessageEventFailure(failure, messages);
+			if(messageEventFailure.isPresent()) {
+				MessageEventFailure theFailure = messageEventFailure.get();
+				String notificationEmailAddress = getErrorNotificationEmailAddress(theFailure);
+				try {
+					String s = this.objectMapper.writeValueAsString(theFailure);
+					boolean validErrorTopic = true;
+					final ProducerRecord<String, String> record = new ProducerRecord<>(errorQueueName, theFailure.getMessage().getId(), s);
+					RecordMetadata metadata;
+					try {
+						metadata = this.producer.send(record).get();
+						logger.debug("Producer Record: key {}, value {}, partition {}, offset {}", record.key(), record.value(),
+								metadata.partition(), metadata.offset());
+					} catch (InterruptedException e) {
+						Thread.currentThread().interrupt();
+						logger.error("Publish message to kafka topic {} failed with an error: {}", errorQueueName, e.getMessage(),
+								e);
+					} catch (ExecutionException e) {
+						if(e.getCause() instanceof TopicAuthorizationException) {
+							validErrorTopic = false;
+							/*
+							 * This exception (although it is an 'authorization' exception, also means that the topic does not exist. 
+							 * Teams do not have to set up an error topic if they don't want to.  
+							 */
+						} else {
+							logger.error("Publish message to kafka topic {} failed with an error: {}", errorQueueName, e.getMessage(), e);
+							throw new ApplicationException(ApplicationException.Code.INTERNAL_ERROR, "Failed to publish the event");
+						}
+					}
+					this.errorEmailSender.newError(this.queueName, validErrorTopic ? errorQueueName : null, notificationEmailAddress, theFailure);
+				} catch(Exception e) {
+					e.printStackTrace();
```

Comment:
If the intention is to swallow the exception, please log it at the `WARN` level.